### PR TITLE
updated changelog

### PR DESCRIPTION
**Pull request by SamMarkowitz**
awaiting final ins and outs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,78 @@
...
+		+ Changed operation hierarchy:
+			+ Python based actions:
+    		```
+    		python_action:
+    		  script
+    		```
+			+ Java based actions:
+    		```
+    		java_action:
+				  gav:
```
Comment (Bonczidai) [inline CHANGELOG.md]: pay attention to indentation

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,78 @@
...
+    		  class_name:
+    		  method_name:
+    		```
+		+ Removed`aggregate` section from parallel steps. Aggregation for a `parallel_loop` is now accomplished in the `publish` section.
+	+ New Validations
+		+ Flow and operation input names must be different than their output names.
+		+ Step arguments and called subflow/operation output names must be different.
```
Comment (Bonczidai) [inline CHANGELOG.md]: I think we call them step inputs, not arguments anymore

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,78 @@
...
+		+ Flow results can no longer contain expressions.
+		+ Step must declares all subflow/operation inputs that are required, not private and don't have a default value.
```
Comment (Bonczidai) [inline CHANGELOG.md]: declare**s**

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,78 @@
...
+		+ All steps must be reachable.
+		+ The `on_failure` section may contain only one step.
+	+ Other Changes
+		+ Changes related to `on_failure`
+			+ Support navigation to `on_failure` by using `on_failure` keyword.
+			+ Navigation to `on_failure` is allowed even if no `on_failure` section exists. In such a case the flow navigates to the `FAILURE` result.
+			+	An `on_failure` step cannot contain a `navigate` section. It always navigates to `FAILURE`.
```
Comment (Bonczidai) [inline CHANGELOG.md]: some nasty spaces?

**Comment by Bonczidai** [comment]
@mihaitusa please take a look on content and see if anything is missing

**Comment by Bonczidai** [comment]
@SamMarkowitz please merge master into this branch :fast_forward:

**Comment by SamMarkowitz** [comment]
@Bonczidai merged and handled your comments.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,78 @@
...
+	        + Utils
+	        + XML
+        + Chef
+        + CI
+	        + CircleCI
+        + Cloud
+	        + Amazon AWS
+		        + Images
+		        + Instances
+		        + Regions
```
Comment (mihaitusa) [inline CHANGELOG.md]: @SamMarkowitz  +Volumes

**Comment by mihaitusa** [comment]
@SamMarkowitz Please take in consideration my last comment.

**Comment by SamMarkowitz** [comment]
@mihaitusa volumes added.

**Comment by Bonczidai** [comment]
thank you guys.. merging

**Pull request merged**